tests/constants.py

```diff
@@ -0,0 +1,45 @@
+SPARQL_QUERY ='''
```

Comment:
IK copy can change so this probably isnt the best approach to just have it here like this, but just testing the tests tbh

Reply:
Yeah we sometimes call data like this "golden data" because it's a specific snapshot of the output of some code. As you probably guessed, if we update the SPARQL query we'd have to update this golden as well.

You don't need to put it in its own file, it's fine to just have it at the top of the test file where it is used.

Reply:
Yeah for sure, will update this next

I agree that it could be put there, but I would argue that it would be ugly so perhaps it shouldn't be put there. When I look at my test file I wanna see tests, not this 40+ lines SPARQL query

Reply:
I understand what you mean. The tradeoff is "cluttering a test where the data is used" versus "cluttering the file system with modules", since you're only going to use this variable in the specific test file. When I look at this `constants.py` file `(which` should probably be called `goldens.py` or `test_data.py`), it's not immediately clear what it is or where it's used. Keeping the test data close to the tests is generally a good practice.

Also, nit, I would name it `EXPECTED_SPARQL_QUERY`, since it is what you expect your function under test to output, not something that is used as a SPARQL query on its own.

Reply:
Fixed